internal/hook/item_hook_handler_test.go

```diff
@@ -703,3 +708,677 @@ func parseLabelSelectorOrDie(s string) labels.Selector {
+					},
+					RestoreHooks: []velerov1api.RestoreResourceHook{
+						{
+							Init: &velerov1api.InitRestoreHook{
+								InitContainers: []corev1api.Container{
+									{
+										Name:    "restore-init-container-1",
+										Image:   "nginx",
+										Command: []string{"a", "b", "c"},
+									},
+									{
+										Name:    "restore-init-container-2",
+										Image:   "nginx",
+										Command: []string{"a", "b", "c"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "shoud not apply any restore hook init containers when resource hook selector mismatch",
+			podInput: corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "app1",
+					Namespace: "default",
+				},
+			},
+			expectedError: nil,
+			expectedPod: &corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "app1",
+					Namespace: "default",
+				},
+			},
+			restoreHooks: []ResourceRestoreHook{
+				{
+					Name: "hook1",
+					Selector: ResourceHookSelector{
+						Namespaces: collections.NewIncludesExcludes().Excludes("default"),
+						Resources:  collections.NewIncludesExcludes().Includes(kuberesource.Pods.Resource),
+					},
+					RestoreHooks: []velerov1api.RestoreResourceHook{
+						{
+							Init: &velerov1api.InitRestoreHook{
+								InitContainers: []corev1api.Container{
+									{
+										Name:    "restore-init-container-1",
+										Image:   "nginx",
+										Command: []string{"a", "b", "c"},
+									},
+									{
+										Name:    "restore-init-container-2",
+										Image:   "nginx",
+										Command: []string{"a", "b", "c"},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "shoud not apply any restore hook init containers when resource hook is nil",
+			podInput: corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "app1",
+					Namespace: "default",
+				},
+			},
+			expectedError: nil,
+			expectedPod: &corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "app1",
+					Namespace: "default",
+				},
+			},
+			restoreHooks: nil,
+		},
+		{
+			name: "shoud not apply any restore hook init containers when resource hook is empty",
+			podInput: corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "app1",
+					Namespace: "default",
+				},
+			},
+			expectedError: nil,
+			expectedPod: &corev1api.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "app1",
+					Namespace: "default",
+				},
+			},
+			restoreHooks: []ResourceRestoreHook{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			handler := InitContainerRestoreHookHandler{}
+			podMap, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&tc.podInput)
+			assert.NoError(t, err)
+			actual, err := handler.HandleRestoreHooks(velerotest.NewLogger(), kuberesource.Pods, &unstructured.Unstructured{Object: podMap}, tc.restoreHooks)
```

Comment:
this err is overwritten before being compared to expectedError

Reply:
Thanks for the catch!